src/expand.ts

```diff
@@ -84,14 +84,95 @@ export interface ExpansionOptions {
   doNotSupportCommands?: boolean;
 }
 
+export function emptyExpansionOptions() {
+  return ({ vars: {}, recursive: true } as ExpansionOptions);
+}
+
+/**
+ * Compute the environment variables that apply with substitutions by opts and expanded_env
+ * @param in_env the base environment to be expaned
+ * @param expanded_env the environment map provided to expand string `${env:var}`
+ *   it's will override the envOverride member of opts
+ * @param intermediateExpand If this is the intermediate expanding
+ * procedure or the final expanding procedure, if it's the
+ * intermediate expanding procedure, the variable that not found
+ * won't not expaned, if it's the final expanding procedure, the variable
+ * that not found would expand to ''.
+ * @param opts Environment expand options
+ */
+export async function computeExpandedEnvironment(
+  in_env: EnvironmentVariables,
+  expanded_env: EnvironmentVariables,
+  intermediateExpand: boolean,
```

Comment:
Can you rename `intermediateExpand` to `ignoreUnknownVars`?  I think that better conveys the meaning.

Reply:
good name, thanks